lib/reportportal.rb

```diff
@@ -1,27 +1,15 @@
 require 'json'
-require 'rest_client'
+require 'http'
 require 'uri'
 require 'pathname'
 require 'tempfile'
 
 require_relative 'report_portal/settings'
-require_relative 'report_portal/patches/rest_client'
 
 module ReportPortal
   TestItem = Struct.new(:name, :type, :id, :start_time, :description, :closed, :tags)
   LOG_LEVELS = { error: 'ERROR', warn: 'WARN', info: 'INFO', debug: 'DEBUG', trace: 'TRACE', fatal: 'FATAL', unknown: 'UNKNOWN' }
 
-  @response_handler = proc do |response, request, result, &block|
-    if (200..207).include? response.code
-      response
-    else
-      p "ReportPortal API returned #{response}"
-      p "Offending request method/URL: #{request.args[:method].upcase} #{request.args[:url]}"
```

Comment:
Does `http` handle errors in the same way?

Reply:
HTTP handles errors in a bit different way. If there is no response (connection error, SSL error, DNS error) an error is raised. If there is a response, its code can be inspected.